src/Updater.php

```diff
@@ -97,71 +104,59 @@ public function logger() {
   /**
    * {@inheritdoc}
    */
-  public function installModules(array $modules) {
-    $successful = TRUE;
+  public function executeUpdate($module, $update_definition_name) {
+    $this->successfulExecute = TRUE;
 
-    foreach ($modules as $module) {
-      try {
-        if ($this->moduleInstaller->install([$module])) {
-          $this->logger->info($this->t('Module @module is successfully enabled.', ['@module' => $module]));
-        }
-        else {
-          $this->logger->warning($this->t('Unable to enable @module.', ['@module' => $module]));
-          $successful = FALSE;
-        }
-      }
-      catch (MissingDependencyException $e) {
-        $this->logger->warning($this->t('Unable to enable @module because of missing dependencies.', ['@module' => $module]));
-        $successful = FALSE;
-      }
+    $update_definitions = $this->configHandler->loadUpdate($module, $update_definition_name);
+    if (isset($update_definitions['__global_actions'])) {
```

Comment:
It feels very Drupal 7 to have an update represented as an array with an API built around the magic __global_actions key

Reply:
Not yet - let's keep the magic here and magic sure it is documented and then maybe followup with something better.

Reply:
I have went a bit further here, so I have added interface for update definition. Later we can provide interface function to in, like - `getInstallModules, getImportConfigs, getExpectedConfig` .. etc. More or less to have some class what will represent YAML file we are generating.

And then we can also extend easier generators of update definition to generate list of new installed modules, imported configs, etc.